src/collection.jl

```diff
@@ -95,3 +95,53 @@ function fitscollection(basedir::String;
     end
     return df
 end
+
+
+"""
+    arrays(df::DataFrame)
+
+Generator for arrays of images of entries in data frame.
+"""
+function arrays end
+
+# generator for image arrays specified by data frames (i.e. path of file, hdu etc.)
+@resumable function arrays(df::DataFrame)
+    for row in eachrow(df)
+        fh = FITS(row.path)
+        try
+            @yield getdata(fh[row.hdu])
+        finally
+            close(fh)
+        end
```

Comment:
Does this work? The more file IO safeguards we can alleviate to other libraries, the better.
```suggestion
        FITS(row.path) do fh
            @yield getdata(fh[row.hdu])
        end
```

If this doesn't work, I'd vote to mimick the other two functions and skip the `try` block (just for consistency).

Reply:
This won't work, this is a bug from `ResumableFunctions.jl`.